Cython/Compiler/Lexicon.py

```diff
@@ -26,6 +26,408 @@ def make_lexicon():
     hexdigit = Any("0123456789ABCDEFabcdef")
     indentation = Bol + Rep(Any(" \t"))
 
+    # Some functions at the end of the file show how to generate these strings.
+    # They're pretty slow to generate at runtime, and require Python3, so
+    # are just included directly here
+    unicode_start_character = (
```

Comment:
I can't find an interface that lets you request all the characters in category. It's certainly possible to iterate character-by-character and use `unicodedata` to look up the category. However both this and my current approach of "iterate through strings trying `isidentifier`) are pretty slow - about 0.5s on my PC. Therefore I don't think this code should be run every time Cython starts, hence the huge constants.

I agree that if a better way exists it'd be good to use it!

Reply:
Thinking some more, since `unicodedata` is regularly updated with new CPython releases, and might learn new identifier characters over time, we should keep this set of characters independent of the specific CPython version that Cython runs with. Meaning, this set should be generated in a controlled way, always with the latest CPython version, and then committed.

BTW, since this is Python code, and the string is generated and guaranteed to consist only of printable characters, why not leave the character decoding to the Python parser and use the real Unicode characters instead of byte strings? (That would make `Lexicon.py` a UTF-8 file, but I'm fine with that.)

Reply:
Also, I would prefer having the identifier character strings as global constants in the module, after the `make_lexicon()` function. That also simplifies their generation a bit. You can use start+end markers like `# GENERATED CODE STARTS HERE` to make them easy to update programmatically. Preferably by a separate script in `bin/`. And if you want to be really ambitious, then you can add a version marker of the CPython version that was used to generate the strings, and let the script print a big warning when they are regenerated with an older version. :)